src/string.cr

```diff
@@ -160,6 +160,16 @@ class String
     end
   end
 
+  def [](str : String)
+    if start = index(str)
+      self[start...start+str.size]
+    end
+  end
+
+  def []?(str : String)
+    !!index(str)
```

Comment:
Yes. Also, is the slice really needed? I would have thought the implementation would be something like this:

```crystal
def [](str : String)
  self[str]?.not_nil! # or raise a better error message
end

def []?(str : String)
  includes?(str) ? str : nil
end
```